drivers/usbdev/cdcncm.c

```diff
@@ -0,0 +1,2525 @@
+/****************************************************************************
+ * drivers/usbdev/cdcncm.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/* References:
+ *   [CDCNCM1.2] Universal Serial Bus - Communications Class - Subclass
+ *               Specification for Ethernet Control Model Devices - Rev 1.2
+ */
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <assert.h>
+#include <debug.h>
+#include <errno.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <string.h>
+#include <time.h>
+
+#include <nuttx/kmalloc.h>
+#include <nuttx/net/netdev_lowerhalf.h>
+#include <nuttx/usb/cdc.h>
+#include <nuttx/usb/cdcncm.h>
+#include <nuttx/usb/usbdev.h>
+#include <nuttx/usb/usbdev_trace.h>
+#include <nuttx/wqueue.h>
+
+#ifdef CONFIG_BOARD_USBDEV_SERIALSTR
+#  include <nuttx/board.h>
+#endif
+
+#include "cdcecm.h"
+
+#ifdef CONFIG_NET_CDCNCM
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/* Work queue support is required. */
+
+#if !defined(CONFIG_SCHED_WORKQUEUE)
```

Comment:
```suggestion
#if !defined(CONFIG_SCHED_WORKQUEUE)
```

why not set NET_CDCNCM depends on SCHED_WORKQUEUE?

Reply:
Done.